syntax = "proto3";

// python -m grpc_tools.protoc -I../web --python_out=. --grpc_python_out=.
// ../web/tunnel.proto python -m grpc_tools.protoc -I../backend --python_out=.
// --grpc_python_out=. ../backend/tunnel.proto create json stringds

package fileharbor;

import "data.proto";
import "common.proto";

service Analyser {
  rpc list_plugins(ListPluginsRequest) returns (ListPluginsReply) {}
  rpc analyse(AnalyseRequest) returns (AnalyseReply) {}
}

message PluginInfo {
  string name = 1;
  string type = 2;
  repeated PluginConfig settings = 3;
}

message ListPluginsRequest {}
message ListPluginsReply { repeated PluginInfo plugin_infos = 1; }

message AnalyseRequest { PluginRun plugin_run = 1; }

message AnalyseReply { repeated PluginResult results = 1; }